### ISSUE TITLE
Scrolling issue when adding section dynamically

### ISSUE THREAD

**Issue by tulsiram-sunrise** [alvarotrigo/react-fullpage#62]
I'm working on a react application where react-fullpage js has been used.

new requirement of adding section(with predefined section simple format, only a text content ) dynamically on a button click, stops it from working properly. 

I'm generating sections for fullpage based on array JSON from props. Adding the new section i'm just adding new array element to JSON array.

If it is not an issue, please suggest me the way to fix this.

**Comment by tulsiram-sunrise** [comment]
Hi @alvarotrigo,
I've generate the codesandbox for the issue got during development.
Have a look at https://codesandbox.io/s/r0ro87ywp4 for the same

**Comment by tulsiram-sunrise** [comment]
Hi @alvarotrigo,
Have you checked the codesandbox?
When i'm adding new section it is getting added to DOM but we can't reach to that section by scrolling.

**Comment by alvarotrigo** [comment]
@tulsiram-sunrise yeap, just checked it. Sorry for the late reply, was on holidays the past weeks.

@cmswalker the issue seems to be related with the following condition on the `componentDidUpdate` function:

```javascript
  componentDidUpdate(prevProps, prevState) {
      console.warn("componentDidUpdate.....................");

    if (prevState.initialized === this.state.initialized) {  // <--- it is entering here and therefore not doing anything
      return;
    }
```
Also related with the dummy section. 
I guess once I add the changes in fullPage.js to get rid of the dummy section requirement we will solve a few issues related with it.

@tulsiram-sunrise  I guess a temporal to bypass the issue would be updating the value of `this.state.initialized` so it won't be the same as the previous state.

**Comment by cmswalker** [comment]
Thanks for finding, will address this soon as it's pretty critical issue. Will likely start putting dynamic sections in the demo too

**Comment by alvarotrigo** [comment]
This issue should be fixed in 0.1.3.

**Comment by tulsiram-sunrise** [comment]
Thanx @alvarotrigo.
